### DISCUSSION TITLE
Recent move to MIT licence and sold/commercial usage

### DISCUSSION THREAD

**Discussion by Romain-Geissler-1A** [rui314/mold#1064] [Q&A] [answered]
Hi,

First thanks for the work you have done with the work behind mold, I have been following from far the different releases and I know that @MaskRay also got your work as inspiration to improve the performances of lld. So I see this as healthy concurrence between lld and mold helping both (and users) in the end ;)

I was wondering about the recent licence move to MIT in release 2.0, and what is the future of sold (and the whole commercial use in general). Basically until sold appeared, I considered mold as something I would definitely give it a try in my company. Then appeared sold and the need to moneytize this linker (and I perfectly understand this need for money, I am definitely not criticizing here). This scared me a bit to be honest, as I know my company, no matter how huge it is, would never pay for a linker (and I also don't like paying for software, instead I think we should contribute to it but... here as well my company doesn't shine... to my great despair...), so I refrained from adopting mold just yet (we are using lld for years). With this new MIT licence, do you confirm that people like me working in private companies and using mold to link proprietary software and making a business with it (with the resulting proprietary binaries, not with mold itself) are "free" (as in allowed) to do it ? And do you foresee a change of licence back again to something more à la sold with a pay-for-use scheme (we only care about Linux, which has never required to pay to use mold, but it may be something you still have in mind as a fallback plan in the future) ?

Cheers,
Romain

**Comment by rui314** [comment, marked as answer]
I have no plan to change the license again. Essentially, changing the license to MIT is a one-way path; if you try to make it proprietary software, someone would fork the project from the last free release.

I'm not a lawyer, and this is not a legal advice. But if I were you, I wouldn't be worried too much.

**Reply by Romain-Geissler-1A** [reply]
Thanks for the quick answer ! (And we have no plan to make any business around linkers/compilers with mold, just to use it to link our own proprietary software, totally unrelated to linkers/compilers). If we try it out and we find issues, for sure we will report it (don't know if we will be able to propose a fix as well, we will see).

**Reply by coffeebe4code** [reply]
The only way you, @Romain-Geissler-1A,  should be worried is if you planned to sell support for, or mold use itself as a commercial offering. At that point you would be a competitor to whichever entity @rui314 would be working with. If you are a user, and writing your own software and linking your proprietary software. This move is a good thing since you could now get commercial support if you ever needed it for your companies mold use.
